internal/expr/bindinputs.go

```diff
@@ -49,41 +35,111 @@ func (te *TypedExpr) BindInputs(args ...any) (pq *PrimedQuery, err error) {
 		if v.Kind() != reflect.Struct && v.Kind() != reflect.Map {
 			return nil, fmt.Errorf("need struct or map, got %s", t.Kind())
 		}
-		if _, ok := typeValue[t]; ok {
+		if _, ok := typeToValue[t]; ok {
 			return nil, fmt.Errorf("type %q provided more than once", t.Name())
 		}
-		typeValue[t] = v
-		typeNames = append(typeNames, t.Name())
-		if !inQuery[t] {
-			// Check if we have a type with the same name from a different package.
-			for _, typeMember := range te.inputs {
-				if t.Name() == typeMember.OuterType().Name() {
-					return nil, fmt.Errorf("parameter with type %q missing, have type with same name: %q", typeMember.OuterType().String(), t.String())
+		typeToValue[t] = v
+	}
+
+	// Generate SQL and query parameters.
+	params := []any{}
+	outputs := []typeinfo.Member{}
+	argTypeUsed := map[reflect.Type]bool{}
+	inCount := 0
+	outCount := 0
+	sqlStr := bytes.Buffer{}
+	for _, te := range *tes {
+		switch te := te.(type) {
+		case *typedInputExpr:
+			typeMember := te.input
+			outerType := typeMember.OuterType()
+			v, ok := typeToValue[outerType]
+			if !ok {
+				return nil, inputMissingError(outerType, typeToValue)
```

Comment:
The error is that one of the types in the type bound expression cannot be found in the input arguments, rather than the other way around. I've changed the names and function description to reflect this better.

Reply:
What about `typeNotBoundError`?

Reply:
This error is now in the `typeinfo` package.